block/retriever.go

```diff
@@ -98,12 +98,10 @@ func (m *Manager) applyLocalBlock(height uint64) error {
 		return fmt.Errorf("validate block from local store: height: %d: %w", height, err)
 	}
 
-	m.retrieverMu.Lock()
```

Comment:
same here. we need a mutex for block application to make sure no 2 blocks can be applied at the same time.